pkg/nodes/enricher/enricher_impl.go

```diff
@@ -92,24 +101,31 @@ func (e *enricherImpl) enrichWithScan(node *storage.Node, nodeInventory *storage
 	}
 
 	for _, scanner := range scanners {
-		if err := e.enrichNodeWithScanner(node, nodeInventory, scanner.GetNodeScanner()); err != nil {
+		log.Debugf("Enriching Node with Scanner %s / %s", scanner.GetNodeScanner().Type(), scanner.GetNodeScanner().Name())
+		if err := e.enrichNodeWithScanner(node, nodeInventory, indexReport, scanner.GetNodeScanner()); err != nil {
 			errorList.AddError(err)
 			continue
 		}
-
 		return nil
 	}
 
 	return errorList.ToError()
 }
 
-func (e *enricherImpl) enrichNodeWithScanner(node *storage.Node, nodeInventory *storage.NodeInventory, scanner types.NodeScanner) error {
+func (e *enricherImpl) enrichNodeWithScanner(node *storage.Node, nodeInventory *storage.NodeInventory, indexReport *v4.IndexReport, scanner types.NodeScanner) error {
 	sema := scanner.MaxConcurrentNodeScanSemaphore()
 	_ = sema.Acquire(context.Background(), 1)
 	defer sema.Release(1)
 
+	var scan *storage.NodeScan
+	var err error
+
 	scanStartTime := time.Now()
-	scan, err := scanner.GetNodeInventoryScan(node, nodeInventory)
+	if scanner.Type() == types.ScannerV4 {
+		scan, err = scanner.GetNodeInventoryScan(node, nil, indexReport)
+	} else {
+		scan, err = scanner.GetNodeInventoryScan(node, nodeInventory, nil)
+	}
```

Comment:
This block is fine, but let's look below it. There are metrics and an error message. Could we make sure that the metrics and the error message capture which scanner has been used for scanning?

Reply:
For the error message, we already capture this in `with scanner %q`.
I have extended the metrics with a new Gauge `node_scan_scanner_version`. 

While writing that I discovered a potential gotcha with the metric for `SetNodeInventoryNumberComponents`:
We're operating on the `inventory`, not the `scan`, so we would miss the v4 statistics. 
Checking, the `scan` does not contain `RhelComponents`, so we would need to change the whole metric in that case. 
WDYT?

Reply:
Then, maybe we should keep the old metric and add a new one next to it that would record the same values but for scanner v4?

Reply:
Planned as [ROX-26474](https://issues.redhat.com/browse/ROX-26474)